syft/pkg/cataloger/deb/parse_dpkg_status.go

```diff
@@ -48,20 +51,54 @@ func parseDpkgStatus(reader io.Reader) ([]pkg.Package, error) {
 }
 
 // parseDpkgStatusEntry returns an individual Dpkg entry, or returns errEndOfPackages if there are no more packages to parse from the reader.
-// nolint:funlen
-func parseDpkgStatusEntry(reader *bufio.Reader) (entry pkg.DpkgMetadata, err error) {
-	dpkgFields := make(map[string]interface{})
+func parseDpkgStatusEntry(reader *bufio.Reader) (pkg.DpkgMetadata, error) {
 	var retErr error
+	dpkgFields, err := extractAllFields(reader)
+	if err != nil {
+		if !errors.Is(err, errEndOfPackages) {
+			return pkg.DpkgMetadata{}, err
+		}
+		retErr = err
+	}
+
+	var entry pkg.DpkgMetadata
+	err = mapstructure.Decode(dpkgFields, &entry)
+	if err != nil {
+		return pkg.DpkgMetadata{}, err
+	}
+
+	name, version := extractSourceVersion(entry.Source)
+	if version != "" {
+		entry.SourceVersion = version
+		entry.Source = name
+	}
```

Comment:
for my clarification only, this means that if version is empty, we don't want to capture the name?

Reply:
good question --I only migrated this part of the code (already existed in this file). Looks like it was added in https://github.com/anchore/syft/pull/297 ... maybe an oversight?

Reply:
I didn't catch that this was migrated over. No strong opinions here, just that this looked odd. If you think this shouldn't get a bit reworked that's fine. If in doubt though, I'd recommend opening up a ticket to look at this later on (no need to block this PR on this)